server/service/client.go

```diff
@@ -172,6 +172,24 @@ func (c *Client) AuthenticatedDo(verb, path, rawQuery string, params interface{}
 	return c.doContextWithHeaders(context.Background(), verb, path, rawQuery, params, headers)
 }
 
+func (c *Client) AuthenticatedDoCustomHeaders(verb, path, rawQuery string, params interface{}, custom_headers map[string]string) (*http.Response, error) {
```

Comment:
To comply with idiomatic Go naming:

```suggestion
func (c *Client) AuthenticatedDoCustomHeaders(verb, path, rawQuery string, params interface{}, customHeaders map[string]string) (*http.Response, error) {
```